### PR TITLE
Various ansible cleanup

### PR DESCRIPTION
**Pull request by emord**
##### SUMMARY
Various deprecation warnings and cleanup I've found over past couple of days

##### ENVIRONMENTS AFFECTED
none

##### ISSUE TYPE
- Change Pull Request

### CODE REVIEW

File: src/commcare_cloud/ansible/deploy_postgres.yml [hunk 1]
```diff
@@ -22,6 +22,7 @@
   roles:
     - {role: pgbouncer, tags: 'pgbouncer'}
 
+# Notes this should be filtered out if the environment does not have remote postgres
```
Comment (emord) [inline src/commcare_cloud/ansible/deploy_postgres.yml]: right now this gets deployed to all servers even if we don't have anything in remote_postgresql @dannyroberts was that intentional?

File: src/commcare_cloud/ansible/roles/java/tasks/main.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ---
 - name: installing repo for Java 8+ in Ubuntu
   apt_repository: repo='ppa:openjdk-r/ppa'
+  when: ansible_distribution_version == '14.04'
```
Comment (emord) [inline src/commcare_cloud/ansible/roles/java/tasks/main.yml]: openjdk-8 is already in bionic https://packages.ubuntu.com/bionic/openjdk-8-jdk so I think this should just be removed

File: src/commcare_cloud/ansible/deploy_postgres.yml (hunk 1 above)
Comment (dannyroberts) [reply src/commcare_cloud/ansible/deploy_postgres.yml]: No that's not intentional, good catch. In it's original version, this only had one task, which ran with an empty `with_items` if there was nothing in `remote_postgresql`. But then on May 20 I added the "Install python-psycopg2 to be able to set up remote postgresql" task to fix the install (something must have previously installed this on django_manage that doesn't anymore), not realizing that that adds an unnecessary install when there's nothing in `remote_postgresql`. If adding `when: groups.remote_postgresql is defined` or the the like to the block works, I support that

File: src/commcare_cloud/ansible/roles/java/tasks/main.yml (hunk 2 above)
Comment (dannyroberts) [reply src/commcare_cloud/ansible/roles/java/tasks/main.yml]: You mean it should be removed once we're officially dropping trusty? I agree with that

**Pull request merged**